src/index.ts

```diff
@@ -11,31 +13,36 @@ type Argv = {
     concurrency: number
 }
 
-yargs
+let app: any = yargs
+    .wrap(yargs.terminalWidth())
     .scriptName('posthog-plugins')
     .option('config', { alias: 'c', describe: 'Config options JSON.', type: 'string' })
-    .option('disable-web', { describe: 'Whether web server should be disabled.', type: 'boolean' })
-    .option('web-port', { alias: 'p', describe: 'Web server port.', type: 'number' })
-    .option('web-hostname', { alias: 'h', describe: 'Web server hostname.', type: 'string' })
-    .option('concurrency', { describe: 'Concurrenct Worker Threads', type: 'number' })
-    .help()
-    .command({
-        command: ['start', '$0'],
-        describe: 'start the server',
-        handler: ({ config, disableWeb, webPort, webHostname, concurrency }: Argv) => {
-            const parsedConfig: PluginsServerConfig = config ? JSON.parse(config) : {}
-            if (typeof webHostname !== 'undefined') {
-                parsedConfig['WEB_HOSTNAME'] = webHostname
-            }
-            if (typeof webPort !== 'undefined') {
-                parsedConfig['WEB_PORT'] = webPort
-            }
-            if (typeof disableWeb !== 'undefined') {
-                parsedConfig['DISABLE_WEB'] = disableWeb
-            }
-            if (typeof concurrency !== 'undefined') {
-                parsedConfig['WORKER_CONCURRENCY'] = concurrency
+
+for (const [key, value] of Object.entries(defaultConfig)) {
+    app = app.option(key.toLowerCase().replaceAll('_', '-'), {
+        describe: `${configHelp[key] || key} [${value}]`,
+        type: typeof value,
+    })
+}
+
+app = app.help().command({
+    command: ['start', '$0'],
+    describe: 'start the server',
+    handler: ({ config, ...otherArgs }: Argv) => {
+        const parsedConfig: Record<string, any> = config ? JSON.parse(config) : {}
+        for (const [key, value] of Object.entries(otherArgs)) {
+            if (typeof value !== 'undefined') {
+                // convert camelCase argument keys to under_score
+                const newKey = key
+                    .replace(/(?:^|\.?)([A-Z])/g, (x, y) => '_' + y.toUpperCase())
+                    .replace(/^_/, '')
+                    .toUpperCase()
+                if (newKey in defaultConfig) {
+                    parsedConfig[newKey] = value
+                }
             }
-            startPluginsServer(parsedConfig, makePiscina)
-        },
-    }).argv
+        }
+        setLogLevel(parsedConfig.LOG_LEVEL || defaultConfig.LOG_LEVEL)
+        startPluginsServer(parsedConfig as PluginsServerConfig, makePiscina)
+    },
+}).argv
```

Comment:
Do we actually need a command ("start")? The server does nothing BUT start, so having to write "start" doesn't add anything.

Reply:
Agreed, removed